IdentityCore/src/parameters/MSIDRequestParameters.m

```diff
@@ -145,7 +145,15 @@ - (void)setCloudAuthorityWithCloudHostName:(NSString *)cloudHostName
     if ([NSString msidIsStringNilOrBlank:cloudHostName]) return;
 
     NSURL *cloudAuthority = [self.authority.url msidAuthorityWithCloudInstanceHostname:cloudHostName];
-    _cloudAuthority = [MSIDAuthorityFactory authorityFromUrl:cloudAuthority context:self error:nil];
+    _internalAuthority= [MSIDAuthorityFactory authorityFromUrl:cloudAuthority context:self error:nil];
```

Comment:
note that this logic has been changed. MSIDAuthorityFactory shouldn't be used anymore, unless we know it's an AAD authority.

Reply:
This could be removed.